.github/workflows/build-test.yml

```diff
@@ -26,6 +51,15 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
+    - name: Potentially override rust version with nightly
```

Comment:
suggestion: replace the body of this repeated boilerplate with a script and a 1-liner invocation

According to the docs for [Github Actions environment variables](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables), `GITHUB_EVENT_NAME` is an environment variable that is available to scripts already.  Even though that doesn't cover `inputs.nightly`, you can still pass that value as an argument to such a script, [as in this example](https://docs.github.com/en/actions/learn-github-actions/contexts#example-usage-of-the-inputs-context-in-a-reusable-workflow).

I'm imagining that such a script would allow you to replace:
```
if: ...
run: |
  ...
```
with just
```
run: cargo make maybe-set-nightly  --inputsNightly=${{ inputs.nightly }}
```

Edit: now there's a second step because there's an orthogonal dimension of the runner OS.  The script + one-liner can still work if you you add an extra CLI arg:

```
run: cargo make maybe-set-nightly  --inputsNightly=${{ inputs.nightly }} --runnerOS=${{ runner.os }}
```

Reply:
Oh, that's a good call, yeah

Reply:
The runner OS is not relevant if we use a script, it's only relevant when using the actions yaml stuff

Reply:
I think I won't change the `if` condition, that seems highly Actions-specific, AND i like that when it doesn't get run the GHA UI shows it greyed out.

I'll just swap out the body.

Reply:
Actually, the annoying thing is that ideally this gets run before cargo-make needs to be installed. Hmm.

Reply:
So the tradeoff is that we must install cargo-make (usually cached) every time we need this. Unfortunate but ... fine I guess?

Reply:
It's _really_ annoying to get the clippy job working.

I'm not convinced this is a net improvement. The GHA files have tons of boilerplate already, perhaps we should write our own Actions package to reduce that, but until we do that it feels like we're not optimizing much, and both cargo-make and Actions are super temperamental so moving it over isn't necessarily a win.

Reply:
Oh i realized I don't need to try to get clippy working.